pmd-apex/src/main/resources/category/apex/design.xml

```diff
@@ -36,6 +36,73 @@ public class Foo {
         </example>
     </rule>
 
+    <rule name="UnusedMethod"
+          language="apex"
+          since="7.0.0"
+          message="Unused methods make understanding code harder"
+          class="net.sourceforge.pmd.lang.apex.rule.design.UnusedMethodRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_apex_design.html#unusedmethod">
+        <description>
+Avoid having unused methods since they make understanding and maintaining code harder.
+
+This rule finds not only unused private methods, but public methods as well.
+
+[ApexLink](https://github.com/nawforce/ApexLink) is used to make this possible and this needs
+additional configuration. The environment variable `PMD_APEX_ROOTDIRECTORY` needs to be set prior to executing
+PMD. With this variable the root directory of the Salesforce metadata, where `sfdx-project.json` resides, is
+specified. ApexLink can then load all the classes in the project and figure out, whether a method is used or not.
+
+For an accurate analysis it is important that the `PMD_APEX_ROOTDIRECTORY` contains a complete set of metadata that
+may be referenced from the Apex source code, such as Custom Objects, Visualforce Pages, Flows and Labels. The
+`PMD_APEX_ROOTDIRECTORY` directory must contain a 'sfdx-project.json', but metadata may be either in the
+[SFDX Source format](https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_source_file_format.htm)
+or the older MDAPI format. The 'packageDirectories' entries in 'sfdx-project.json' are used to determine which
+directories to search for metadata, if a .forceignore file is present it will be honoured.
+
+If the Apex code references external packages via namespace(s) you should declare these in your 'sfdx-project.json'
+file using the 'plugins' syntax below to avoid errors:
+<![CDATA[
```

Comment:
I think we should surround this code snippet with "\`\`\`json" ... "\`\`\`" to have it rendered nicely in the rule docs.

Reply:
I can fix this when I merge... probably tonight